src/vue/align-editor/align-editor.vue

```diff
@@ -36,45 +37,75 @@ export default {
   data () {
     return {
       defineAlignShow: false,
-      alignments: [],
-      currentAlignment: {},
-      alignmentId: 0,
-      prevClickedWordType: null
+      prevClickedWordType: null,
+      showAlignment: [],
+      originUpdated: 1,
+      targetUpdated: 1
     }
   },
   watch: {
     async showEditor () {
       this.defineAlignShow = true
+      this.updateOriginEditor()
+      this.updateTargetEditor()
     }
   },
   computed: {
     defineAlignShowLabel () {
       return this.defineAlignShow ? 'hide' : 'show'
     },
     showAlignEditor () {
-      return this.sourceText && this.sourceText.text && this.translationText && this.translationText.text
+      return this.originAlignedText && this.originAlignedText.tokens && this.targetAlignedText && this.targetAlignedText.tokens
+    },
+    originAlignedText () {
+      return this.originUpdated ? this.$textC.originAlignedText : {}
+    },
+    targetAlignedText () {
+      return this.targetUpdated ? this.$textC.targetAlignedText : {}
     }
   },
   methods: {
+    updateOriginEditor () {
+      this.originUpdated = this.originUpdated + 1
+    },
+
+    updateTargetEditor () {
+      this.targetUpdated = this.targetUpdated + 1
+    },
+
     toggleDefineAlignShow () {
       this.defineAlignShow = !this.defineAlignShow
     },
-    clickWord (textWord) {
-      console.info('clickWord ', textWord)
-      if (textWord.textType === 'source' && (!this.prevClickedWordType || this.prevClickedWordType === 'source')) {
-        this.finishCurrentAlignment()
-        this.startNewAlignment(textWord)
+    clickWord (token) {
+      if (token.textType === 'origin' && (!this.prevClickedWordType || this.prevClickedWordType !== 'origin')) {
+        this.finishCurrentAlignmentGroup()
+        this.startNewAlignmentGroup(token)

```

Comment:
ok - will work on this change

Reply:
big change

Reply:
another note on the highlighting that I think I missed in the requirements doc:

Once a token belongs to saved alignment group, it should be displayed a different color than unaligned tokens so that you can easily see which tokens have been aligned and which have not.

Reply:
ok - will think about it